cmd/migration-manager/internal/cmds/batch.go

```diff
@@ -105,27 +107,85 @@ func (c *cmdBatchAdd) Run(cmd *cobra.Command, args []string) error {
 		Name: args[0],
 	}
 
-	b.Target, err = c.global.Asker.AskChoice(fmt.Sprintf("What target should be used by this batch? (Choices: %q) ", strings.Join(targets, "', '")), targets, "")
+	if len(targets) == 1 {
+		b.Target = targets[0]
+		fmt.Printf("Using target %q\n", b.Target)
+	} else {
+		defaultTargetHint := "(" + strings.Join(targets, ", ") + "): "
+		b.Target, err = c.global.Asker.AskChoice("What target should this batch use? "+defaultTargetHint, targets, "")
+		if err != nil {
+			return err
+		}
+	}
+
+	resp, err := c.global.doHTTPRequestV1("/targets/"+b.Target, http.MethodGet, "", nil)
+	if err != nil {
+		return err
+	}
+
+	var t api.Target
+	err = responseToStruct(resp, &t)
+	if err != nil {
+		return err
+	}
+
+	it, err := target.NewInternalIncusTargetFrom(t)
+	if err != nil {
+		return err
+	}
+
+	// Connect to the target.
+	err = it.Connect(cmd.Context())
```

Comment:
So I think that's going to be a problem. In general we're now moving to the assumption that the client only has access to migration-manager and any needed authentication server. They may not have direct access to the source or target so can't directly make API calls to get things like the project or storage pool list.

The ongoing work from @gibmat is going to remove whatever remaining direct interactions the client has, instead having all source and target operations be handled by the server.

Reply:
Right, it makes sense to keep all source/target communication on the daemon side. In that case should I remove populating choices for now until we have a formal mechanism for fetching target data from the daemon?

Reply:
Yeah, probably best, we'll have to just ask for the raw value for now.

Reply:
Done